### PR TITLE
Grating solver

### PR DESCRIPTION
**Pull request by jordiferrero**
### Description of the change
Added a solver for the grating/spectrometer equation. 
- A `utils.axis` function that converts an axis in pixel/undefined units to wavelength in nanometers
- A `px_to_nm_grating_solver` in the `LumiSpectrum` class that converts the object's signal axis to wavelength in nanometers.

### Progress of the PR
- [x] Change implemented
- [x] Add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import lumispy as lum
import numpy as np
s = LumiSpectrum(np.ones(20),))
s.px_to_nm_grating_solver(*exp_paramers , inplace=True)
s.axes_manager.signal_axes[0].units

>> 'nm'
```

### CODE REVIEW

**Comment by jordiferrero** [comment]
I do have a few comments/questions:
1. What is the best way to "inherit" docstrings? Let me explain. In this PR I have 2 functions, one stored in `utils.axis` and the other one is the method that calls the `util` function though the `LumiSpectrum` class. Both functions have the same parameters (`*exp_parameters`). They are described manually in both functions. Is there an easy way to inherit the docstrings for the `utils` function onto the `LumiSpectrum` function?

2. I am assuming that the refraction index in air is 1, but that is not "exact". Do you think I should not assume so? 

Thank you!

**Comment by jlaehne** [comment]
>     1. What is the best way to "inherit" docstrings? Let me explain. In this PR I have 2 functions, one stored in `utils.axis` and the other one is the method that calls the `util` function though the `LumiSpectrum` class. Both functions have the same parameters (`*exp_parameters`). They are described manually in both functions. Is there an easy way to inherit the docstrings for the `utils` function onto the `LumiSpectrum` function?
Good point, would apply to #56 as well.

>     2. I am assuming that the refraction index in air is 1, but that is not "exact". Do you think I should not assume so?
Well it should be easy to use the function `_n_air(wl)` to correct for that (for the range 185 nm to 1700 nm). Why not use it even if the improvement is probably not so great. The error introduced by the abberation of the spectrometer will probably be larger, but that is another story.

File: lumispy/signals/luminescence_spectrum.py [hunk 1]
```diff
@@ -340,6 +340,58 @@ def remove_background_from_file(self, background=None, inplace=False, **kwargs):
                 return self.map(lambda s, bkg: s - bkg, bkg=bkg_y, inplace=True)
 
 
+    def px_to_nm_grating_solver(self, gamma_deg, deviation_angle_deg, focal_length_mm, ccd_width_mm,
+                           grating_central_wavelength_nm, grating_density_gr_mm, inplace=False,):
```
Comment (jlaehne) [inline lumispy/signals/luminescence_spectrum.py:344]: parameter names are fairly long, on the other hand they are very explicit, which is nice

File: lumispy/signals/luminescence_spectrum.py [hunk 2]
```diff
@@ -340,6 +340,58 @@ def remove_background_from_file(self, background=None, inplace=False, **kwargs):
...
+
+        """
+
+        nm_axis = solve_grating_equation(self.axes_manager.signal_axes[0],
+                                         gamma_deg, deviation_angle_deg, focal_length_mm, ccd_width_mm,
+                                         grating_central_wavelength_nm, grating_density_gr_mm)
+
+        # in place conversion
+        if inplace:
+            self.axes_manager.remove(-1)
```
Comment (jlaehne) [inline lumispy/signals/luminescence_spectrum.py:383]:
```
if inplace: s = self
else: s = self.deepcopy()

s.axes_manager.remove(-1)
s.axes_manager._axes.append(nm_axis)
return s
```

saves you a few duplicate lines

**Comment by jordiferrero** [comment]
@jlaehne Most of these "aesthetic" changes I did not add them myself manually... 
I use PyCharm and it always corrects for them automatically, to keep things consistent, I suppose...
I personally find it confusing to not use espaces in some occasions while using them most of the time.

Thanks for the comments though!

File: lumispy/signals/luminescence_spectrum.py [hunk 3]
```diff
@@ -378,20 +378,21 @@ def px_to_nm_grating_solver(self, gamma_deg, deviation_angle_deg, focal_length_m
...
-            s.axes_manager.remove(-1)
-            s.axes_manager._axes.append(nm_axis)
+
+        s.axes_manager.remove(-1)
+        s.axes_manager._axes.append(nm_axis)
+
+        if inplace:
+            return
+        else:
             return s
```
Comment (jlaehne) [inline lumispy/signals/luminescence_spectrum.py:392]: `return s` is sufficient in this case, as it is a pointer to self in case of `inplace`

**Comment by ericpre** [comment]
> 1. What is the best way to "inherit" docstrings? Let me explain. In this PR I have 2 functions, one stored in `utils.axis` and the other one is the method that calls the `util` function though the `LumiSpectrum` class. Both functions have the same parameters (`*exp_parameters`). They are described manually in both functions. Is there an easy way to inherit the docstrings for the `utils` function onto the `LumiSpectrum` function?

Include string in the docstring, for the syntax see for example:
https://github.com/hyperspy/hyperspy/tree/RELEASE_next_minor/hyperspy/docstrings
and 
https://github.com/hyperspy/hyperspy/blob/165e407e777b503cdbc1b3d29dcf481ac15b7e4b/hyperspy/signal.py#L2866-L2901

**Comment by jlaehne** [comment]
Thanks @ericpre, I already used that in an update to #56 @jordiferrero. Though I preferred to keep the docstring in the file where the function is defined rather than in an extra folder.

**Comment by ericpre** [comment]
Yes, the string used by the docstring can live anywhere! I guess the reason why the docstring are in separate folder in hyperspy is that most of them are used across several files.

**Comment by jordiferrero** [comment]
I have now included the changes to the dosctrings.
Ready for final review.

**Review by jlaehne — APPROVED** [review]
LGTM

**Pull request merged**